src/main/java/games/strategy/engine/framework/map/download/DownloadMapsWindow.java

```diff
@@ -265,6 +265,10 @@ private static ListSelectionListener createDescriptionPanelUpdatingSelectionList
       final JList<String> gamesList, final List<DownloadFileDescription> maps, final MapAction action,
       final JLabel mapSizeLabelToUpdate) {
     return e -> {
+      if (e.getValueIsAdjusting()) {
+        return;
+      }
+
```

Comment:
A `JList` model may fire a chain of `ListSelectionEvent`s to support transition states that may be rendered during, for example, drag-and-drop.  The last event in the chain will return `false` from `getValueIsAdjusting()` and represents the final state of the list selection change.  This is the only event we should be processing to ensure multiple requests to get download length for the same URI are not fired concurrently.